test/e2e/handler/nodes_test.go

```diff
@@ -33,6 +34,12 @@ var _ = Describe("Nodes", func() {
 				interfacesNameForNodeEventually(node).Should(ContainElement(primaryNic))
 			}
 		})
+		It("should able to read NMState resources with cluster-reader user", func() {
+			Eventually(func() error {
+				_, err := cmd.Kubectl("get", "nns", "--as=system:serviceaccount:default:test-user")
```

Comment:
You don't need to call kubectl, you can use the controller-runtime client instancianting it with Impersonate options https://pkg.go.dev/k8s.io/client-go/rest#Config

Reply:
As we spoke, it would require another client configured as impersonate in order to keep the kube:admin client intact,
and doesn't seem to worth it, we can recalculate later if needed